TestFoundation/FTPServer.swift

```diff
@@ -0,0 +1,275 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//This is a very rudimentary FTP server written plainly for testing URLSession FTP Implementation.
+import Dispatch
+
+#if DEPLOYMENT_RUNTIME_OBJC || os(Linux)
+    import Foundation
+    import Glibc
+    import XCTest
+#else
+    import CoreFoundation
+    import SwiftFoundation
+    import Darwin
+    import SwiftXCTest
+#endif
+
+class _FTPSocket {
+
+    private var listenSocket: Int32!
+    private var socketAddress = UnsafeMutablePointer<sockaddr_in>.allocate(capacity: 1)
+    private var socketAddress1 = UnsafeMutablePointer<sockaddr_in>.allocate(capacity: 1)
+    private var connectionSocket: Int32!
+    var dataSocket: Int32! // data socket for communication
+    var dataSocketPort: UInt16! // data socket port, should be sent as part of header
+    private func isNotNegative(r: CInt) -> Bool {
```

Comment:
Bad name, based on the code it should be `isNotMinusOne()`

Reply:
Actually this method implementation is same as the ```isNotNegative``` method in HTTPServer .. Do you want me to rename that also ?